template.go

```diff
@@ -299,7 +300,7 @@ func writeDebianRules(dir string, pkgType packageType) error {
 	fmt.Fprintf(f, "#!/usr/bin/make -f\n")
 	fmt.Fprintf(f, "\n")
 	fmt.Fprintf(f, "%%:\n")
-	fmt.Fprintf(f, "\tdh $@ --builddirectory=_build --buildsystem=golang\n")
+	fmt.Fprintf(f, "\tdh $@ --builddirectory=debian/build --buildsystem=golang\n")
```

Comment:
NACK. There are already too many packages using that path.

Reply:
This is not an API. There isn't any downside for new packages to use this
clearly cleaner and better way, and old packages are trivial to update, and
testing that the update works correctly is also trivial. The different
gitignore/clean has been used before should not be an argument on its own.
Otherwise no progress could happen, right?

Reply:
Maybe we should drop this flag entirely and update https://salsa.debian.org/go-team/packages/dh-golang/-/blob/81892fe8d3a51a22c910a649052669dd9f18837e/lib/Debian/Debhelper/Buildsystem/golang.pm#L275-279 instead?